# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: mozglue/baseprofiler/core/shared-libraries-linux.cc
# Commit: 32b940c88cca
# Full Hash: 32b940c88ccade4477f59b9b9c71049ac0d6b061
# Author: Nazım Can Altınova <canaltinova@gmail.com>
# Date: 2024-09-17 09:28:38
# Regressor Bug: 1634785
# File Overlap Count: 1
# Description:
#   Bug 1634785 - Fix how shared library names are found on linux and macOS r=mstange,profiler-reviewers
#   
#   While testing the de-duplication on Linux and macOS, I noticed that the
#   symbolication was completely broken due to having incorrect library names in
#   libs array. They were always like: "<obj-dir>/dist/bin/libxul.so" instead of
# ==============================================================================

diff -r f9b6ae065ffc -r 32b940c88cca mozglue/baseprofiler/core/shared-libraries-linux.cc
--- a/mozglue/baseprofiler/core/shared-libraries-linux.cc	Mon Sep 16 21:09:32 2024 +0000
+++ b/mozglue/baseprofiler/core/shared-libraries-linux.cc	Mon Sep 16 21:09:32 2024 +0000
@@ -696,7 +696,7 @@
                                          unsigned long offset = 0) {
   std::string pathStr = path;
 
-  size_t pos = pathStr.rfind('\\');
+  size_t pos = pathStr.rfind('/');
   std::string nameStr =
       (pos != std::string::npos) ? pathStr.substr(pos + 1) : pathStr;
 